lib/active_support/cache/database_store.rb

```diff
@@ -81,7 +89,9 @@ def increment(name, amount = 1, options = nil)
 
         entry = Entry.new(amount, **options.merge(version: normalize_version(name, options)))
         expires_at = Time.zone.at(entry.expires_at) if entry.expires_at
-        attrs = { key: normalize_key(name, options), value: Marshal.dump(entry.value), version: entry.version.presence, expires_at: expires_at }
+        # Integer and float entries do not warrant compression so that upsert remains possible with a DB-drive increment
+        attrs = { key: normalize_key(name, options), compression: 'plain', value: compress(entry.value, 'plain'), version: entry.version.presence,
```

Comment:
pls revert, use `entry_attributes` (let's merge the other PR first and rebase this one)

Reply:
Done!